### ISSUE TITLE
Can't use regex in selector-type-no-unknown ignoreTypes

### ISSUE THREAD

**Issue by feimosi** [stylelint/stylelint#2254]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A bug where entering proper JavaScript regex into `selector-type-no-unknown: ignoreTypes` fails.

> Which rule, if any, is this issue related to?

`selector-type-no-unknown: ignoreTypes`

> What CSS is needed to reproduce this issue?

Sass:

```scss
&:at-least(2) {
  float: left;
  width: 50%;
  ...
}
```

> What stylelint configuration is needed to reproduce this issue?

I've tried various regex values like `/^\d/`, `/^[1-9]/`, `/^2/` etc.

```json
{
    "selector-type-no-unknown": [true,
      "ignoreTypes": [
        ["/^\\d/"]
      ]
    ],
}
```

> Which version of stylelint are you using?

6.8.0

> How are you running stylelint: CLI, PostCSS plugin, Node API?

PostCSS with `gulp-postcss`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes

> What did you expect to happen?

Not to flag `&:at-least(2)` as an error.

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
xxx/LogIn.scss
⚠  Invalid value "/^2/" for option "ignoreTypes" of rule "selector-type-no-unknown" [stylelint]
```

```shell
xxx/Actions.scss
⚠  Invalid value "/^\d/" for option "ignoreTypes" of rule "selector-type-no-unknown" [stylelint]
```

**Comment by jeddy3** [comment]
@feimosi Thanks for using the issue template.

I think two things might be in play here:

1. Your old version of stylelint.
2. Your configuration object.

Please update to `7.7.1`. Then change your configuration object to:

```json
{
  "rules": {
    "selector-type-no-unknown": [true, {
      "ignoreTypes": ["/^[1-9]/"]
    }],
  }
}
```

As secondary options are _objects_.

[Tests look good](https://github.com/stylelint/stylelint/blob/master/lib/rules/selector-type-no-unknown/__tests__/index.js#L135-L163).

**Comment by feimosi** [comment]
The configuration object was indeed written in a wrong way by my mistake. 
But upgrading stylelint only solved the problem. 

Btw. it's not nice to close an issue, before I get a chance to reply.

**Comment by davidtheclark** [comment]
> it's not nice to close an issue, before I get a chance to reply

@feimosi: It's pretty common practice for large repositories that deal with a lot of issues, so please don't feel offended. People are still notified when you post follow-up comment on closed issues, so the discussion can continue and the issue can always be re-opened.

**Comment by feimosi** [comment]
Never heard of such practice. What other big projects use it? You should give at least a week to reply. IMHO closing means closing.
